### PR TITLE
Add support for TLS authentication on Gusite LDAP

### PR DESCRIPTION
**Pull request by debojitkakoti**
Add support for TLS authentication.
Add support for Gsuite LDAP.

Signed-off-by: Debojit Kakoti <debojkakoti88@gmail.com>

#### :heavy_check_mark: Checklist

- [ :heavy_check_mark:] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [:heavy_check_mark: ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/catalog-backend-module-ldap/config.d.ts [hunk 1]
```diff
@@ -56,6 +56,8 @@ export interface Config {
       tls?: {
         // Node TLS rejectUnauthorized
         rejectUnauthorized?: boolean;
+        keys: string;
```
Comment (jhaals) [inline plugins/catalog-backend-module-ldap/config.d.ts:59]: looks like this is going to be breaking for people running with `rejectUnauthorized` today. Can we perhaps put them in a sub section or make them optional?

**Review by jhaals — CHANGES_REQUESTED** [review]
Thanks for the PR, looks like there's a merge conflict currently. Some tests would be great to add 🙏

File: plugins/catalog-backend-module-ldap/config.d.ts (hunk 1 above)
Comment (debojitkakoti) [reply plugins/catalog-backend-module-ldap/config.d.ts:59]: @jhaals I am not much familiar with NodeJs, but I will try my best for writing tests !

File: plugins/catalog-backend-module-ldap/src/processors/LdapOrgEntityProvider.ts [hunk 2]
```diff
@@ -89,6 +89,11 @@ export interface LdapOrgEntityProviderOptions {
    * The function that transforms a group entry in LDAP to an entity.
    */
   groupTransformer?: GroupTransformer;
+
+  /**
+   * Specify which vendor to use.
+   */
+  vendorOverride?: String;
```
Comment (freben) [inline plugins/catalog-backend-module-ldap/src/processors/LdapOrgEntityProvider.ts:96]: All string types should be lowercase `string`. But would it make more sense to be able to pass in an actual vendor instance here instead? Then the package could export

```ts
export const vendors = Object.freeze({
  microsoft: (): LdapVendor => ...,
});
```

that way people can easily add their completely own ones before waiting for us to do it in open source and then waiting for a release to happen

**Comment by freben** [comment]
Ping @debojitkakoti :)

**Pull request closed without merging**